src/components/hooks/useElementDimensions.js

```diff
@@ -0,0 +1,42 @@
+import {useEffect, useRef, useState} from 'react';
+import {ResizeObserver as Polyfill} from '@juggle/resize-observer';
+
+/**
+ * Custom hook that returns an element reference and the dimensions of the
+ * element it refers to.
+ *
+ * @returns {Array} Element reference and dimensions in pixels
+ */
+export default function useElementDimensions() {
+  const elementRef = useRef({});
+  const [height, setHeight] = useState(0);
+  const [width, setWidth] = useState(0);
+
+  useEffect(() => {
+    const ResizeObserver = window.ResizeObserver || Polyfill;
```

Comment:
why this or that? can you explain a bit what this means? brief one line comment perhaps?

Reply:
Sure, I can add a comment. This checks to see if the browser has a native implementation of `ResizeObserver`, and uses that instead of the polyfill